File: swift/task3.swift
import Foundation

func digitSum(_ n: Int) -> Int {
    return String(n).compactMap { Int(String($0)) }.reduce(0, +)
}

func digitProduct(_ n: Int) -> Int {
    return String(n).compactMap { Int(String($0)) }.reduce(1, *)
}

print("Введите числа через пробел:", terminator: " ")
if let input = readLine() {
    let numbers = input.split(separator: " ").compactMap { Int($0) }
    var result: [Int] = []

    for (i, num) in numbers.enumerated() {
        if digitSum(num) < digitProduct(num) {
            result.append(i + 1) // позиция с 1
        }
    }

    print("Номера, где сумма цифр < произведения: \(result.map { String($0) }.joined(separator: " "))")
}

File: c#/task3.cs
using System;
using System.Linq;

class Task3 {
    static int DigitSum(int n) {
        return n.ToString().Sum(c => c - '0');
    }

    static int DigitProduct(int n) {
        return n.ToString().Aggregate(1, (prod, c) => prod * (c - '0'));
    }

    static void Main() {
        Console.Write("Введите числа через пробел: ");
        var numbers = Console.ReadLine().Split().Select(int.Parse).ToList();

        Console.Write("Номера подходящих чисел: ");
        for (int i = 0; i < numbers.Count; i++) {
            int num = numbers[i];
            if (DigitSum(num) < DigitProduct(num)) {
                Console.Write((i + 1) + " ");
            }
        }
        Console.WriteLine();
    }
}
